### PR TITLE
feat(IdentityCenter): add Identity Center user resource and instance data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add Identity Center user resource and instance data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add Identity Center user resource and instance data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
go test ./huaweicloud/services/acceptance/identitycenter -v -run TestAccDatasourceIdentityCenter_basic -timeout 360m
=== RUN   TestAccDatasourceIdentityCenter_basic
=== PAUSE TestAccDatasourceIdentityCenter_basic
=== CONT  TestAccDatasourceIdentityCenter_basic
--- PASS: TestAccDatasourceIdentityCenter_basic (7.39s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/identitycenter    12.356s

go test ./huaweicloud/services/acceptance/identitycenter -v -run TestAccIdentityCenterUser_basic -timeout 360m
=== RUN   TestAccIdentityCenterUser_basic
=== PAUSE TestAccIdentityCenterUser_basic
=== CONT  TestAccIdentityCenterUser_basic
--- PASS: TestAccIdentityCenterUser_basic (32.06s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/identitycenter    37.223s
```

### CODE REVIEW

File: docs/data-sources/identitycenter_instance.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+---
+
+# huaweicloud_identitycenter_instance
+
+Use this data source to get the Identity Center instance info.
+
+## Example Usage
+
+```hcl
+data "huaweicloud_identitycenter_instance" "test"{}
```
Comment (ShiChangkuo) [inline docs/data-sources/identitycenter_instance.md]:
```suggestion
data "huaweicloud_identitycenter_instance" "system"{}
```

File: docs/resources/identitycenter_user.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+---
+subcategory: "IAM Identity Center"
+---
+
+# huaweicloud_identitycenter_user
+
+Manages an IdentityCenter user resource within HuaweiCloud.
```
Comment (ShiChangkuo) [inline docs/resources/identitycenter_user.md]:
```suggestion
Manages an Identity Center user resource within HuaweiCloud.
```

File: docs/resources/identitycenter_user.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+changes should be applied to the IdentityCenter user, or the resource definition should be updated to align with the
+instance. Also, you can ignore changes as below.
+
+```
+resource "huaweicloud_identitycenter_user" "instance_1" {
+  ...
+
+  lifecycle {
+    ignore_changes = [
+      "identity_store_id", password_mode",
```
Comment (ShiChangkuo) [inline docs/resources/identitycenter_user.md]: with out ""
```suggestion
      identity_store_id, password_mode,
```

File: huaweicloud/services/acceptance/identitycenter/data_source_huaweicloud_identitycenter_instance_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceIdentityCenter_basic(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckOutput("id_is_not_empty", "true"),
+					resource.TestCheckOutput("identity_store_id_is_not_empty", "true"),
+					resource.TestCheckOutput("urn_is_not_empty", "true"),
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/identitycenter/data_source_huaweicloud_identitycenter_instance_test.go]: use **resource.TestCheckResourceAttrSet** will be better

File: huaweicloud/services/acceptance/identitycenter/resource_huaweicloud_identitycenter_user_test.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+func getIdentityCenterUserResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getIdentityCenterUser: Query Identity Center user
+	var (
+		getIdentityCenterUserHttpUrl = "v1/identity-stores/{identity_store_id}/users/{user_id}"
+		getIdentityCenterUserProduct = "identitystore"
+	)
+	getIdentityCenterUserClient, err := cfg.NewServiceClient(getIdentityCenterUserProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating IdentityCenter Client: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/identitycenter/resource_huaweicloud_identitycenter_user_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating Identity Center client: %s", err)
```

please modify others

File: huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_user.go [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+		CreateContext: resourceIdentityCenterUserCreate,
+		UpdateContext: resourceIdentityCenterUserUpdate,
+		ReadContext:   resourceIdentityCenterUserRead,
+		DeleteContext: resourceIdentityCenterUserDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceIdentityCenterUserImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_user.go:37]: region is useless?

File: huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_user.go [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+	if err != nil {
+		return diag.Errorf("error deleting IdentityCenterUser: %s", err)
+	}
+
+	return nil
+}
+
+func resourceIdentityCenterUserImportState(_ context.Context, d *schema.ResourceData,
+	_ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), "/", 2)
```
Comment (ShiChangkuo) [inline huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_user.go]: use *strings.Split* will be more suitable.

File: docs/data-sources/identitycenter_instance.md (hunk 1 above)
Comment (houpeng80) [reply docs/data-sources/identitycenter_instance.md]: fixed

File: docs/resources/identitycenter_user.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/identitycenter_user.md]: done

File: docs/resources/identitycenter_user.md (hunk 3 above)
Comment (houpeng80) [reply docs/resources/identitycenter_user.md]: done

File: huaweicloud/services/acceptance/identitycenter/data_source_huaweicloud_identitycenter_instance_test.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/identitycenter/data_source_huaweicloud_identitycenter_instance_test.go]: fixed

File: huaweicloud/services/acceptance/identitycenter/resource_huaweicloud_identitycenter_user_test.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/identitycenter/resource_huaweicloud_identitycenter_user_test.go]: done

File: huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_user.go (hunk 7 above)
Comment (houpeng80) [reply huaweicloud/services/identitycenter/resource_huaweicloud_identitycenter_user.go]: fixed

**Comment by houpeng80** [comment]
go test ./huaweicloud/services/acceptance/identitycenter -v -run TestAccDatasourceIdentityCenter_basic -timeout 360m
=== RUN   TestAccDatasourceIdentityCenter_basic
=== PAUSE TestAccDatasourceIdentityCenter_basic
=== CONT  TestAccDatasourceIdentityCenter_basic
--- PASS: TestAccDatasourceIdentityCenter_basic (7.33s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/identitycenter    12.402s

go test ./huaweicloud/services/acceptance/identitycenter -v -run TestAccIdentityCenterUser_basic -timeout 360m
=== RUN   TestAccIdentityCenterUser_basic
=== PAUSE TestAccIdentityCenterUser_basic
=== CONT  TestAccIdentityCenterUser_basic
--- PASS: TestAccIdentityCenterUser_basic (12.84s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/identitycenter    17.806s

File: docs/resources/identitycenter_user.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason. The missing attributes include: `identity_store_id` and `password_mode`.
+It is generally recommended running `terraform plan` after importing an IdentityCenter user. You can then decide if
+changes should be applied to the IdentityCenter user, or the resource definition should be updated to align with the
+instance. Also, you can ignore changes as below.
+
+```
+resource "huaweicloud_identitycenter_user" "instance_1" {
```
Comment (ShiChangkuo) [inline docs/resources/identitycenter_user.md]:
```suggestion
```hcl
resource "huaweicloud_identitycenter_user" "user_1" {
```

Comment (houpeng80) [reply docs/resources/identitycenter_user.md]: done

**Pull request merged**